src/ctap/pin_protocol_v1.rs

```diff
@@ -644,21 +651,25 @@ mod test {
         persistent_store.set_pin_hash(&pin_hash);
     }
 
-    fn encrypt_standard_pin(shared_secret: &[u8; 32]) -> Vec<u8> {
+    // Fails on PINs bigger than 64 byte..
+    fn encrypt_pin(shared_secret: &[u8; 32], pin: Vec<u8>) -> Vec<u8> {
+        assert!(pin.len() <= 64);
+        let mut padded_pin = [0u8; 64];
+        padded_pin[..pin.len()].copy_from_slice(&pin[..]);
         let aes_enc_key = crypto::aes256::EncryptionKey::new(shared_secret);
         let mut blocks = [[0u8; 16]; 4];
-        blocks[0][0] = 0x31;
-        blocks[0][1] = 0x32;
-        blocks[0][2] = 0x33;
-        blocks[0][3] = 0x34;
+        let (b0, b1, b2, b3) = array_refs!(&padded_pin, 16, 16, 16, 16);
+        blocks[0][..].copy_from_slice(b0);
+        blocks[1][..].copy_from_slice(b1);
+        blocks[2][..].copy_from_slice(b2);
+        blocks[3][..].copy_from_slice(b3);
         let iv = [0u8; 16];
         cbc_encrypt(&aes_enc_key, iv, &mut blocks);
+        blocks.iter().flatten().cloned().collect::<Vec<u8>>()
+    }
 
-        let mut encrypted_pin = Vec::with_capacity(64);
-        for b in &blocks {
-            encrypted_pin.extend(b);
-        }
-        encrypted_pin
+    fn encrypt_standard_pin(shared_secret: &[u8; 32]) -> Vec<u8> {
```

Comment:
Question: what is a "standard" PIN? The string "1234"? This function would be worth a 1-line documentation.

Reply:
Done.